components/brave_origin/brave_origin_policy_manager.cc

```diff
@@ -112,6 +113,49 @@ PoliciesEnabledMap BraveOriginPolicyManager::GetAllProfilePolicies(
   return policies;
 }
 
+void BraveOriginPolicyManager::SetBrowserPolicyValue(std::string_view pref_name,
+                                                     bool value) {
+  SetPolicyValueInternal(pref_name, value, browser_policy_definitions_,
+                         /*profile_id=*/std::nullopt);
+}
+
+void BraveOriginPolicyManager::SetProfilePolicyValue(
+    std::string_view pref_name,
+    bool value,
+    std::string_view profile_id) {
+  CHECK(!profile_id.empty())
+      << "Profile ID cannot be empty for profile policies";
+
+  SetPolicyValueInternal(pref_name, value, profile_policy_definitions_,
+                         profile_id);
+}
+
+void BraveOriginPolicyManager::SetPolicyValueInternal(
+    std::string_view pref_name,
+    bool value,
+    const BraveOriginPolicyMap& policy_definitions,
+    std::optional<std::string_view> profile_id) {
+  CHECK(initialized_) << "BraveOriginPolicyManager not initialized";
+  CHECK(local_state_) << "BraveOriginPolicyManager local state should exist";
+
+  // Check if this is a valid pref
+  const BraveOriginPolicyInfo* policy_info =
+      base::FindOrNull(policy_definitions, pref_name);
+  if (!policy_info) {
+    LOG(ERROR) << "Unknown " << (profile_id.has_value() ? "profile" : "browser")
```

Comment:
Is this error log permanent? If so, should we change it to `VLOG` to avoid noise?